src/main/java/com/farion/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -19,12 +22,18 @@
 @RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final CategoryService categoryService;
     private final BookMapper bookMapper;
     private final BookSpecificationBuilder bookSpecificationBuilder;
 
     @Override
     public BookDto createBook(CreateBookRequestDto requestDto) {
         Book book = bookMapper.toModel(requestDto);
+        book.getCategories().forEach(category -> {
+            CategoryDto categoryDto = categoryService.getById(category.getId());
+            category.setName(categoryDto.name());
+            category.setDescription(categoryDto.description());
+        });
```

Comment:
do not use for each, that's 100 requests to DB per 100 categories, use `repo.findAllByIdIn(ids)`
then compare the size of the dto collection and resulted repo call collection - throw ex if mismatch, and save. You do not need to set a name and description of each category by hand having a valid ID is enough to make a relation and it'll be fine